server/v2/appmanager/appmanager.go

```diff
@@ -21,10 +22,44 @@ type AppManager[T transaction.Tx] struct {
 	exportState func(ctx context.Context, dst map[string]io.Writer) error
 	importState func(ctx context.Context, src map[string]io.Reader) error
 
-	stf stf.STFI[T]
+	stf *stf.STF[T]
```

Comment:
Doesn't it prohibit people to implement their own stf?

Reply:
Yes, @testinginprod authored this change in https://github.com/cosmos/cosmos-sdk/pull/19927 so deferred to his judgement there.

Reply:
we should go back to interface for loose coupling.

Reply:
it's more about the location rather than the interface, stf defining the interface and providing the implementation is not that idiomatic, ideally the consumer defines the interface it needs: app manager should define the subset of methods it needs from STF in such a way that it can be replaced (if needed)

Reply:
yea we can leave a todo to make the interface in appmanager, realised the lower down